tests/tests/test_state_scripts.py

```diff
@@ -632,6 +764,139 @@ class TestStateScripts(MenderTesting):
         "ArtifactFailure_Error_55", # Error for this state doesn't exist, should never run.
     ]
 
+    @pytest.mark.usefixtures("standard_setup_one_client_bootstrapped")
+    @pytest.mark.parametrize("test_set", REBOOT_TEST_SET)
+    def test_reboot_recovery(self, test_set):
+        if not env.host_string:
+            execute(
+                self.test_reboot_recovery,
+                test_set,
+                hosts=get_mender_clients())
+            return
+
+        client = env.host_string
+        work_dir = "test_state_scripts.%s" % client
+
+        script_content = '#!/bin/sh\n\necho "$(basename $0)" >> /data/test_state_scripts.log\n'
+        # Even though this script is actually run twice, only show it once in the logs
+        script_stop_mender_on_entry = '#!/bin/sh\ntest "$(grep -c "$(basename $0)" /data/test_state_scripts.log)" -eq 0 && echo "$(basename $0)" >> /data/test_state_scripts.log && systemctl stop mender\nexit 0\n'
+        script_failure_content = script_content + 'sync\necho b > /proc/sysrq-trigger\n' # flush to disk before killing
+        script_reboot_once = '#!/bin/sh\ntest "$(grep -c "$(basename $0)" /data/test_state_scripts.log)" -eq 0 && echo "$(basename $0)" >> /data/test_state_scripts.log && sync && echo b > /proc/sysrq-trigger\necho "$(basename $0)" >> /data/test_state_scripts.log\nexit 0'
```

Comment:
Could you break up these lines a bit? It's hard to read.

Reply:
Why are there commands after the sysrq call? They will never get executed, will they? Is it to make sure of that?

Reply:
It will be rerun in the case of the device dying in commit_leave.

Reply:
Ah right, I missed the conditional there. Another reason to break this up a bit for easier reading. :-)